optuna_dashboard/ts/components/TrialTable.tsx

```diff
@@ -91,55 +87,38 @@ export const TrialTable: FC<{
       }))
     columns.push(...objectiveColumns)
   }
-  if (
-    studyDetail?.union_search_space.length ===
+  const isDynamicSpace =
+    studyDetail?.union_search_space.length !==
     studyDetail?.intersection_search_space.length
-  ) {
-    studyDetail?.intersection_search_space.forEach((s) => {
-      const sortable = s.distribution.type !== "CategoricalDistribution"
-      const filterChoices =
-        s.distribution.type === "CategoricalDistribution"
-          ? s.distribution.choices.map((c) => c.value)
-          : undefined
-      columns.push({
-        field: "params",
-        label: `Param ${s.name}`,
-        toCellValue: (i) =>
-          trials[i].params.find((p) => p.name === s.name)
-            ?.param_external_value || null,
-        sortable: sortable,
-        filterChoices: filterChoices,
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        less: (firstEl, secondEl, _): number => {
-          const firstVal = firstEl.params.find(
-            (p) => p.name === s.name
-          )?.param_internal_value
-          const secondVal = secondEl.params.find(
-            (p) => p.name === s.name
-          )?.param_internal_value
-
-          if (firstVal === secondVal) {
-            return 0
-          } else if (firstVal && secondVal) {
-            return firstVal < secondVal ? 1 : -1
-          } else if (firstVal) {
-            return -1
-          } else {
-            return 1
-          }
-        },
-      })
-    })
-  } else {
+  studyDetail?.union_search_space.forEach((s) => {
+    const sortable = s.distribution.type !== "CategoricalDistribution"
+    const filterChoices =
+      s.distribution.type === "CategoricalDistribution"
+        ? s.distribution.choices.map((c) => c.value)
+        : undefined
+    const filterChoicesWithNull: (string | null)[] | undefined = filterChoices
+      ? [...filterChoices, null]
+      : undefined
     columns.push({
       field: "params",
-      label: "Params",
+      label: `Param ${s.name}`,
       toCellValue: (i) =>
-        trials[i].params
-          .map((p) => p.name + ": " + p.param_external_value)
-          .join(", "),
+        trials[i].params.find((p) => p.name === s.name)?.param_external_value ||
+        null,
+      sortable: sortable,
+      filterChoices: isDynamicSpace ? filterChoicesWithNull : filterChoices,
```

Comment:
`isDynamicSpace` is calculated for each study, but `filterChoices` is set for each column. This `isDynamicSpace` should be implemented as the value whether the column is dynamic space.

```
import optuna


def objective(trial):
    category = trial.suggest_categorical("category", ["foo", "bar"])
    if category == "foo":
        return (trial.suggest_float("x1", 0, 10) - 2) ** 2
    else:
        return -((trial.suggest_float("x2", -10, 0) + 5) ** 2)


study = optuna.create_study(storage="sqlite:///test.db")
study.optimize(objective, n_trials=100)
```

In the above code, this study's `isDynamicSpace` is always true because some of columns are dynamic spaces. As a result,  `filterChoices` will be `filterChoicesWithNull` even though `category` is not dynamic space.

Reply:
Or, something like this?

```
const filterChoices =
      s.distribution.type === "CategoricalDistribution"
        ? s.distribution.choices.map((c) => c.value)
        : undefined
values = trials[i].params.find((p) => p.name === s.name)?.param_external_value
if values.contain(null):
    filterChoices.append(null)
```